pycloudlib/azure/instance.py

```diff
@@ -264,6 +264,11 @@ def add_network_interface(self) -> str:
 
         Returns the private ip address of the new NIC.
         """
+        # pylint: disable=too-many-locals
+        us = datetime.datetime.now().strftime("%f")
+        # get ip address object
+        ip_address_obj = self._create_ip_address()
+        ip_config_name = f"{self.name}-{us}-ip-config"
```

Comment:
Are we doing this twice? It looks like this is peformed 20 links below as well

Reply:
Wrong merge conflicts resolution, amended.